src/TurbulenceClosures/turbulence_closure_implementations/scale_invariant_smagorinsky.jl

```diff
@@ -0,0 +1,330 @@
+#####
+##### The turbulence closure proposed by Smagorinsky and Lilly.
+##### We also call this 'Constant Smagorinsky'.
+#####
+using Oceananigans.Operators: volume
+using Oceananigans.Grids: XBoundedGrid, YBoundedGrid, ZBoundedGrid, XYBoundedGrid, XZBoundedGrid, YZBoundedGrid, XYZBoundedGrid
+
+abstract type AveragingProcedure end
```

Comment:
abstract types are always written with the prefix `Abstract`, eg `AbstractAveragingProcedure`